eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java

```diff
@@ -138,18 +142,150 @@ private int optimized(final ForeignTojo tojo, final Optimization optimization)
      * Common optimization for all tojos.
      * @return Optimization for all tojos.
      */
-    private Optimization optimization() {
-        final Optimization opt;
+    private Function<XML, XML> optimization() {
+        final Function<XML, XML> opt;
         final Train<Shift> train = this.measured(new TrParsing());
         if (this.trackOptimizationSteps) {
-            opt = new OptSpy(
+            opt = new OptimizeMojo.OptSpy(
                 train,
                 this.targetDir.toPath().resolve(OptimizeMojo.STEPS)
             );
         } else {
-            opt = new OptTrain(train);
+            opt = new OptimizeMojo.OptTrain(train);
         }
         return opt;
     }
 
+    /**
+     * Optimization that spies.
+     * @since 0.68.0
+     */
+    public static final class OptSpy implements Function<XML, XML> {
```

Comment:
@ArtemGet can this class be package private?

Reply:
Yes, because used only inside package.